support-workers/src/main/scala/com/gu/support/workers/lambdas/CreatePaymentMethod.scala

```diff
@@ -60,14 +60,15 @@ class CreatePaymentMethod(servicesProvider: ServiceProvider = ServiceProvider)
       state.acquisitionData
     )
 
-  def createStripePaymentMethod(stripe: StripePaymentFields, currency: Currency, stripeService: StripeService): Future[CreditCardReferenceTransaction] =
-    stripeService
-      .createCustomer(stripe.stripeToken, currency)
-      .map { stripeCustomer =>
-        val card = stripeCustomer.source
-        CreditCardReferenceTransaction(card.id, stripeCustomer.id, card.last4,
-          CountryGroup.countryByCode(card.country), card.exp_month, card.exp_year, card.zuoraCardType)
-      }
+  def createStripePaymentMethod(stripe: StripePaymentFields, stripeService: StripeService#WithCurrency): Future[CreditCardReferenceTransaction] =
+    (stripe match {
+      case StripeSourcePaymentFields(source) => CreateCustomer.fromSource(source)
+      case StripePaymentMethodPaymentFields(paymentMethod) => CreateCustomer.fromPaymentMethod(paymentMethod)
+    })(stripeService).map { stripeCustomer =>
+      val card = stripeCustomer.source
+      CreditCardReferenceTransaction(card.id, stripeCustomer.id, card.last4,
```

Comment:
here I'm assuming the existing card.id will be alright for Zuora's primary token field.
I think zuora only uses it when it does a POST on /v1/charges https://stripe.com/docs/api/charges/create
and zuora actually posts it as "card" instead of "source" because they are assuming a version of stripe API from 2015.  This may be ignored by stripe, but I have a feeling that stripe let you get away with it, as we are actually making the same mistake in support-workers!  Anyway, we only have a single (default) source so I don't think there will be any problem regardless.

Reply:
just to update this comment - this assumption was incorrect, we need to pass the payment_method id instead, and use a stripe v2 gateway, but it does seem to work in sandbox.